spec/js-api/options.d.ts

```diff
@@ -110,23 +110,28 @@ export interface Options<sync extends 'sync' | 'async'> {
   verbose?: boolean;
 }
 
-/** Additional options specific to compiling from a string. */
-export type StringOptions<sync extends 'sync' | 'async'> = Options<sync> & {
+export interface StringOptionsWithoutImporter<sync extends 'sync' | 'async'>
```

Comment:
should this type have `importer?: never`  to ensure that passing an `importer` option gets typed as `StringOptionsWithImporter` and reports a missing `url` if needed.

Reply:
It turns out this doesn't work because `StringOptionsWithImporter` extends `StringOptionsWithoutImporter`. But that superclass relationship should provide pretty decent error messages on its own.